common/configdir.c

```diff
@@ -22,15 +22,22 @@ static char *default_configdir(const tal_t *ctx)
 	return path;
 }
 
+static char *default_rpc_file(const tal_t *ctx)
+{
+	char *path;
+	path = path_basename(ctx, "lightning-rpc");
+	return path;
+}
```

Comment:
Ah!  I see, thanks for bug report!

--rpc-file= was trying to free the old string, but it was a literal.  The simplest fix here is to change below: